src/liquid/components/ld-file-upload/ld-choose-file/ld-choose-file.tsx

```diff
@@ -0,0 +1,303 @@
+import {
+  Component,
+  h,
+  Host,
+  Prop,
+  Element,
+  State,
+  Event,
+  EventEmitter,
+  Fragment,
+} from '@stencil/core'
+import { getClassNames } from '../../../utils/getClassNames'
+import { getAssetPath } from '../../../utils/assetPath'
+import type { UploadItem } from '../ld-file-upload'
+
+let chooseFileCount = 0
+
+/**
+ * @virtualProp ref - reference to component
+ */
+@Component({
+  tag: 'ld-choose-file',
+  styleUrl: 'ld-choose-file.css',
+  shadow: true,
+  assetsDirs: ['assets'],
+})
+export class LdChooseFile {
+  @Element() el: HTMLLdChooseFileElement
+  private idPrefix = `ld-choose-file-${++chooseFileCount}`
+
+  /** Max. file size in bytes */
+  @Prop() maxSize?: number
+
+  /** Defines whether upload starts immediately after choosing files or after confirmation. */
+  @Prop() startUpload?: boolean = false
+
+  /** Defines whether start upload button has been clicked while startUpload = false */
+  @Prop() startUploadClicked?: boolean = false
+
+  /** Defines whether selection of multiple input files is allowed. */
+  @Prop() selectMultiple?: boolean = false
+
+  /** Defines whether the total progress of all uploading files will be shown in the progress button */
+  @Prop() showProgress?: boolean = false
```

Comment:
Do we need this? Why don't we just show the progress if we have the information of the items?